cdap-data-fabric/src/main/java/io/cdap/cdap/store/RepositoryStore.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.store;
+
+import com.google.inject.ImplementedBy;
+import io.cdap.cdap.common.NamespaceNotFoundException;
+import io.cdap.cdap.common.RepositoryNotFoundException;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryMeta;
+
+/**
+ * Store for repository configuration. This interface is by default implemented by {@link DefaultRepositoryStore},
+ * unless explicitly overridden for testing.
+ */
+@ImplementedBy(DefaultRepositoryStore.class)
```

Comment:
avoid injecting like this unless there is a good reason. Use normal guice bindings instead.

Though I also don't see multiple implementations, why does this need to be an interface instead of a concrete class?

Reply:
removed `@ImplementedBy`. Added the interface and the Default implementation cause I referred to `NamespaceStore` and `DefaultNamespaceStore`. Seems there's no good reason to do so other than just the concrete implementation

Reply:
if there's only one implementation we can remove the interface and just have the concrete class. A lot of the existing code doesn't follow the best practices, especially older code like the NamespaceStore.